src/com/gfredericks/schpec/numbers.clj

```diff
@@ -0,0 +1,119 @@
+(ns com.gfredericks.schpec.numbers
+  (:require [clojure.spec :as s]
+            [clojure.spec.gen :as gen])
+  (:import [java.math BigDecimal MathContext RoundingMode]))
+
+(def finite?
+  "Returns true if the given value is an actual finite number"
+  (let [falsies #{Double/POSITIVE_INFINITY
+                  Double/NEGATIVE_INFINITY
+                  Double/NaN}]
+    (fn [x]
+      (and (number? x)
+           (not (contains? falsies x))))))
+
+(s/def ::finite
+  (s/spec finite?
+          :gen #(gen/double* {:infinite? false :NaN? false})))
```

Comment:
Good question. If we wanted the latter, we'd need to choose the generator from, what, longs, doubles, bigdecs, bigints? Would that be sufficiently broad, or would we want shorts and ints and floats also?